Commit Message: Add section how to create commit message with subject (#28655)
TheOdinProject/curriculum: git/foundations_git/commit_messages.md

- Avoid using vague commit messages such as "saved" or "updated".
- Commit early and often!

> #### How to create commit with subject and body in message

Reviewer: Nit
```suggestion
#### How to commit with a subject and body in the message
```

> Up until now, you are told to commit with `git commit -m <message>`. To make a commit with subject and body in message, the simplest way is to type `git commit` without the m flag and message argument.

Reviewer: Nit
```suggestion
Up until now, you've been told to commit with `git commit -m <message>`. To make a commit with subject and body in message, the simplest way is to type `git commit` without the `-m` flag and message argument.
```

> Up until now, you are told to commit with `git commit -m <message>`. To make a commit with subject and body in message, the simplest way is to type `git commit` without the m flag and message argument.
> Doing so opens a new Visual Code tab if [Visual Code is set as Git editor](https://www.theodinproject.com/lessons/foundations-git-basics#changing-the-git-commit-message-editor). You can remove comments and enter your multi-line messages. When you close the tab, Git creates your commit with that commit message.

Reviewer: Single line breaks won't render any differently to a single space. A blank line will render them as separate paragraphs.
```suggestion
Up until now, you are told to commit with `git commit -m <message>`. To make a commit with subject and body in message, the simplest way is to type `git commit` without the m flag and message argument.

Doing so opens a new Visual Code tab if [Visual Code is set as Git editor](https://www.theodinproject.com/lessons/foundations-git-basics#changing-the-git-commit-message-editor). You can remove comments and enter your multi-line messages. When you close the tab, Git creates your commit with that commit message.
```

> Doing so opens a new Visual Code tab if [Visual Code is set as Git editor](https://www.theodinproject.com/lessons/foundations-git-basics#changing-the-git-commit-message-editor). You can remove comments and enter your multi-line messages. When you close the tab, Git creates your commit with that commit message.

Reviewer: Nit suggestion
```suggestion
Doing so will open a new Visual Studio Code tab if you had [set Visual Studio Code as the Git editor](https://www.theodinproject.com/lessons/foundations-git-basics#changing-the-git-commit-message-editor). You can remove any comments and enter your multi-line messages. When you save and close the tab, your commit will be created.
```

> 

Reviewer: I think we can put this section right before "When to commit", as it directly addresses how to do the things taught before. It can then be made a level 3 heading instead of level 4 heading.

Author: Make sense. I moved it and change heading level
